crates/crypto/src/key/key.rs

```diff
@@ -1,8 +1,8 @@
 use ssi_jwk::JWK;
 use ssi_jws::Error as JWSError;
 
-/// Enum defining all supported cryptographic key types.
-pub enum KeyType {
+/// Enum defining all supported cryptographic algorithms for a [`Key`].
+pub enum KeyAlgorithm {
```

Comment:
The values for this enum appear to me to be Elliptic Curves (as defined in https://www.rfc-editor.org/rfc/rfc7518.html#section-6.2.1.1 and who's possible values are in the registry located on https://www.iana.org/assignments/jose/jose.xhtml#web-key-elliptic-curve). How about we call this `Curve`?

Reply:
supporting RSA 😄

Reply:
Curious what thought are here, specially from @mistermoe and @decentralgabe